Development/src/pages/devices/DevicesShow.js

```diff
@@ -0,0 +1,133 @@
+import JsonIcon from '../../components/JsonIcon';
+import MapTags from '../../components/TagsField';
+import TAIField from '../../components/TAIField';
+import UrlField from '../../components/URLField';
+import QueryVersion from '../../components/QueryVersion';
+import ChipConditionalLabel from '../../components/ChipConditionalLabel';
+
+const cookies = new Cookies();
+
+const DevicesTitle = ({ record }) => {
+    return (
+        <span>
+            Device:{' '}
+            {record
+                ? record.label
+                    ? `${record.label}`
+                    : `${record.id}`
+                : 'Unknown'}
+        </span>
+    );
+};
+
+const DevicesShowActions = ({ basePath, data, resource }) => (
+    <TopToolbar title={<DevicesTitle />}>
+        {data ? (
+            <Button
+                label={'Raw'}
+                href={cookies.get('Query API') + '/' + resource + '/' + data.id}
+                title={'View raw'}
+            >
+                <JsonIcon />
+            </Button>
+        ) : null}
+        <ListButton title={'Return to ' + basePath} basePath={basePath} />
+    </TopToolbar>
+);
+
+const DevicesShow = props => {
+    const controllerProps = useShowController(props);
+    return (
+        <ShowView
+            {...props}
+            {...controllerProps}
+            title={<DevicesTitle />}
+            actions={<DevicesShowActions />}
+        >
+            <SimpleShowLayout>
+                <TextField label="ID" source="id" />
+                <TAIField source="version" />
+                <TextField source="label" />
+                {controllerProps.record && QueryVersion() >= 'v1.1' && (
+                    <TextField source="description" />
+                )}
+                {controllerProps.record && QueryVersion() >= 'v1.1' && (
+                    <FunctionField
+                        label="Tags"
+                        render={record =>
+                            Object.keys(record.tags).length > 0
+                                ? MapTags(record)
+                                : null
+                        }
+                    />
+                )}
+                <hr />
+                <TextField source="type" />
+                {controllerProps.record && QueryVersion() >= 'v1.1' && (
+                    <ArrayField source="controls">
+                        <Datagrid>
+                            <UrlField source="href" label="Address" />
+                            <TextField source="type" />
+                        </Datagrid>
+                    </ArrayField>
+                )}
+                <ReferenceField
+                    label="Node"
+                    source="node_id"
+                    reference="nodes"
+                    link="show"
+                >
+                    <ChipConditionalLabel source="label" />
+                </ReferenceField>
+                <ReferenceArrayField source="receivers" reference="receivers">
```

Comment:
Are all the attributes that have gone from these fields actually the defaults, then?